AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/AWSIncomingEventReconciliationQueue.swift

```diff
@@ -29,15 +47,31 @@ final class AWSIncomingEventReconciliationQueue: IncomingEventReconciliationQueu
                 continue
             }
             reconciliationQueues[modelName] = queue
+
+            let cancellable = queue.publisher.sink(receiveCompletion: { completed in
```

Comment:
nit: I generally dislike (although I'm certainly guilty of it) inlining closure code in the init method. Can we add handlers for the sink completion & value?

Reply:
Agree!  Came up with this.  I think it's pretty decent:
```
let modelReconciliationQueueSink = queue.publisher.sink(receiveCompletion: onReceiveCompletion(completed:),
                                                        receiveValue: onRecieveValue(receiveValue:))
```

then:
```
private func onReceiveCompletion(completed: Subscribers.Completion<DataStoreError>) {
...
}
```
```
private func onRecieveValue(receiveValue: ModelReconciliationQueueEvent) {
...
}
```

way cleaner! thanks!

I will also apply this to other places as well.